File: Assets/ProjectGame/Lib/data/enum/GameObject_Character/GameObject_CharacterID.py
from enum import IntEnum
from typing import Callable, List, Any


class GameObject_CharacterID(IntEnum):
    NONE = 0  # デフォルト値（C#のNoneに相当）
    Prefab = 1  # GameObject_Character_Prefab
    MAX = 2

def to_int(id: "GameObject_CharacterID") -> int:
    return int(id)


def to_GameObject_CharacterID(id: int) -> "GameObject_CharacterID":
    return GameObject_CharacterID(id)


def to_index(id: "GameObject_CharacterID") -> int:
    return int(id) - 1


def for_id(action: Callable[["GameObject_CharacterID"], None]):
    if action is None:
        raise ValueError("action cannot be None")
    start = GameObject_CharacterID.Prefab.value
    for i in range(start, GameObject_CharacterID.MAX.value):
        try:
            value = GameObject_CharacterID(i)
            action(value)
        except ValueError:
            continue  # 未定義の値はスキップ


def find_all(predicate: Callable[["GameObject_CharacterID"], bool]) -> List["GameObject_CharacterID"]:
    if predicate is None:
        raise ValueError("predicate cannot be None")
    results: List["GameObject_CharacterID"] = []
    start = GameObject_CharacterID.Prefab.value
    for i in range(start, GameObject_CharacterID.MAX.value):
        try:
            value = GameObject_CharacterID(i)
            if predicate(value):
                results.append(value)
        except ValueError:
            continue
    return results


def find(predicate: Callable[["GameObject_CharacterID"], bool]) -> "GameObject_CharacterID":
    if predicate is None:
        raise ValueError("predicate cannot be None")
    start = GameObject_CharacterID.Prefab.value
    for i in range(start, GameObject_CharacterID.MAX.value):
        try:
            value = GameObject_CharacterID(i)
            if predicate(value):
                return value
        except ValueError:
            continue
    return GameObject_CharacterID.NONE

File: Assets/ProjectGame/Lib/data/enum/GameObject_Character/GameObject_CharacterID.js
// GameObject_CharacterID - Pure JavaScript Version
export const GameObject_CharacterID = {
    None: 0,  // デフォルト値（C#互換）
    Prefab: 1,  // GameObject_Character_Prefab
    Max: 2
};

export const GameObject_CharacterIDExtensions = {
    /**
     * Enumを数値に変換
     */
    toInt(id) {
        return Number(id);
    },

    /**
     * 数値をEnumに変換
     */
    toGameObject_CharacterID(id) {
        return id;
    },

    /**
     * 0-based indexに変換
     */
    toIndex(id) {
        return Number(id) - 1;
    },

    /**
     * すべてのIDに対して処理を実行
     */
    forID(action) {
        if (typeof action !== 'function') {
            throw new Error('action must be a function');
        }
        const start = GameObject_CharacterID.Prefab;
        const max = GameObject_CharacterID.Max;
        for (let id = start; id < max; id++) {
            if (Object.values(GameObject_CharacterID).includes(id)) {
                action(id);
            }
        }
    },

    /**
     * 条件に合うすべてのIDを返す
     */
    findAll(predicate) {
        if (typeof predicate !== 'function') {
            throw new Error('predicate must be a function');
        }
        const results = [];
        const start = GameObject_CharacterID.Prefab;
        const max = GameObject_CharacterID.Max;
        for (let id = start; id < max; id++) {
            if (Object.values(GameObject_CharacterID).includes(id)) {
                if (predicate(id)) {
                    results.push(id);
                }
            }
        }
        return results;
    },

    /**
     * 条件に合う最初のIDを返す（見つからなければ None）
     */
    find(predicate) {
        if (typeof predicate !== 'function') {
            throw new Error('predicate must be a function');
        }
        const start = GameObject_CharacterID.Prefab;
        const max = GameObject_CharacterID.Max;
        for (let id = start; id < max; id++) {
            if (Object.values(GameObject_CharacterID).includes(id)) {
                if (predicate(id)) {
                    return id;
                }
            }
        }
        return GameObject_CharacterID.None;
    }
};
